File: gpio4j/src/main/java/org/overworld/gpio4j/serial/wire/Deconfig.java
package org.overworld.gpio4j.serial.wire;

import java.io.ByteArrayInputStream;
import java.io.ByteArrayOutputStream;
import java.io.IOException;

import org.overworld.gpio4j.serial.LineByte;
import org.overworld.gpio4j.serial.MessageType;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.With;

@AllArgsConstructor
@NoArgsConstructor
@With
@Data
public class Deconfig implements Message {
	
	private Integer pin;

	@Override
	public Deconfig parse(ByteArrayInputStream wireData) throws IOException {
		
		if (wireData.available() < 2) throw new IllegalStateException("Incorrect number of bytes");
		
		pin = new LineByte(wireData.readNBytes(2)).asInt();
		return this;
	}

	@Override
	public ByteArrayOutputStream wireData() {
		
		if (pin == null) throw new IllegalStateException("Pin is not set when generating wire data");
		
		ByteArrayOutputStream result = new ByteArrayOutputStream();
		result.writeBytes(new LineByte(pin).getHiLo());
		return result;
	}

	@Override
	public MessageType getType() {
		return MessageType.DECONFIG;
	}
}

File: gpioBoardDriver/Deconfig.cpp
#include "Deconfig.h"

class Deconfig : public Message {
public:
    Deconfig() : pin(0) {}
    Deconfig(int pin) : pin(pin) {}

    Deconfig* parse(std::istream& wireData) override {
        if (wireData.rdbuf()->in_avail() < 2) throw std::runtime_error("Incorrect number of bytes");

        std::vector<uint8_t> bytes(2);
        wireData.read(reinterpret_cast<char*>(bytes.data()), 2);
        pin = LineByte(bytes).asInt();
        return this;
    }

    std::vector<uint8_t> wireData() override {
        if (!pin) throw std::runtime_error("Pin is not set when generating wire data");

        LineByte lineByte(pin);
        return lineByte.getHiLo();
    }

    MessageType getType() override {
        return MessageType::DECONFIG;
    }

private:
    int pin;
};
